src/cell.cpp

```diff
@@ -616,19 +705,18 @@ void CSGCell::to_hdf5_inner(hid_t group_id) const
   if (!region_.empty()) {
     std::stringstream region_spec {};
     for (int32_t token : region_) {
-      if (token == OP_LEFT_PAREN) {
-        region_spec << " (";
-      } else if (token == OP_RIGHT_PAREN) {
-        region_spec << " )";
-      } else if (token == OP_COMPLEMENT) {
-        region_spec << " ~";
-      } else if (token == OP_INTERSECTION) {
-      } else if (token == OP_UNION) {
-        region_spec << " |";
-      } else {
+      if (token < OP_UNION) {
         // Note the off-by-one indexing
         auto surf_id = model::surfaces[abs(token) - 1]->id_;
         region_spec << " " << ((token > 0) ? surf_id : -surf_id);
+      } else if (token > OP_COMPLEMENT) {
+        if (token == OP_LEFT_PAREN) {
+          region_spec << " (";
+        } else {
+          region_spec << " )";
+        }
+      } else if (token == OP_UNION) {
+        region_spec << " |";
```

Comment:
Are these changes really needed? It looks to me like the previous code would have worked fine?

Reply:
No, they are not, I changed it when I was debugging. We can get rid of the complement if statement as we've already done De Morgan's laws.

Reply:
Yeah, let's just revert to the previous logic here. I would actually just keep the `if` block for complement since then the `else` block at the bottom definitely refers to a non-operator.